lib/llvm/Makefile

```diff
@@ -0,0 +1,64 @@
+# Build llvm library and optionally clang and compiler-rt
+# Based on [yurydelendik wasmllvm](https://gist.github.com/yurydelendik/4eeff8248aeb14ce763e)
+
+ROOT_DIR := $(shell pwd)
+
+$(warning MAKECMDGOALS=$(MAKECMDGOALS) LLVM_PROJ=$(LLVM_PROJ))
+
+ifeq ($(MAKECMDGOALS),llvm-6.0.0)
+  LLVM_PROJ := llvm-6.0.0
+else ifeq ($(MAKECMDGOALS),llvm-3.9.1)
+  LLVM_PROJ := llvm-3.9.1
+else
+  LLVM_PROJ := llvm-3.9.1
+  MAKECMDGOALS += llvm-3.9.1
+endif
+
+LLVM_SRCDIR := $(ROOT_DIR)/src-$(LLVM_PROJ)
+LLVM_BUILDDIR := $(ROOT_DIR)/build
+
+BUILD_ENGINE := "Unix Makefiles"
+BUILD_TYPE := Release
+LLVM_INSTALLDIR := $(ROOT_DIR)/dist
+LINKER := bfd
+
+ifeq ($(BUILD_ENGINE),Ninja)
```

Comment:
im confused, both ninja and make are supported?

if they are both supported, what is the reason to support ninja and not just make?

Reply:
Ninja is much faster when rebuilding a code base and I've found it advantageous in the past. It defaults to make, but I can remove it if you'd like.

Reply:
Can you quantify that?